knowledge_verificator/backend.py

```diff
@@ -2,47 +2,128 @@
+    response.status_code = 200
+    return format_response(data=material_db.materials)
 
 
 @endpoints.get('/materials/{material_id}')
-def get_material(material_id: int, q: Union[str, None] = None):
+def get_material(material_id: str, response: Response):
     """
     Get a specific learning material.
 
     Args:
-        material_id (int): ID of a material to retrieve.
-        q (Union[str, None], optional): Query to find a material if
-        `material_id` is not known. Defaults to None.
+        material_id (str): ID of a material to retrieve.
 
     Returns:
-        dict: Response with status of a request
-        status and a learning material if request was processed correctly.
+        dict: Under `data` key, there are `material_id` and `material` keys.
     """
-    data = {'material_id': material_id, 'query': q}
-    return format_response(data=data, successful=True)
+    try:
+        material = material_db[material_id]
+    except KeyError:
+        message = f'Material with id = {material_id} was not found.'
+        response.status_code = 404
+        return format_response(message=message)
+
+    data = {'material_id': material_id, 'material': material}
+    response.status_code = 200
+    return format_response(data=data)
+
+
+@endpoints.post('/materials')
+def add_material(material: Material, response: Response) -> dict:
+    """
+    Endpoint to add a learning material to a database.
+
+    Args:
+        material (Material): Learning material to be added.
+        response (Response): Response to a request. Automatically passed.
+
+    Returns:
+        dict: Under 'data' key, there is `material_id` key containing ID
+        of the newly added material.
+    """
+    response.status_code = 200
+    message = ''
+    try:
+        material_db.add_material(material=material)
+    except ValueError as e:
+        message = str(e)
+        response.status_code = 400
+    except FileExistsError as e:
+        message = str(e)
+        response.status_code = 403
+
+    if response.status_code != 200:
+        return format_response(message=message)
+
+    data = {'material_id': material.id}
+    return format_response(data=data)
 
 
 @endpoints.delete('/materials/{material_id}')
-def delete_material(material_id: int):
-    """Delete a learning material with the supplied `material_id`."""
-    return format_response(data=str(material_id), successful=False)
+def delete_material(material_id: str, response: Response) -> dict:
+    """
+    Endpoint to delete a learning material.
+
+    Args:
+        material_id (str): ID of the material to be removed.
+        response (Response): Response to a request. Automatically passed.
+
+    Returns:
+        dict: Under `data` key, there is `material_id` key containing ID
+        of the removed material.
+    """
+    material_db.delete_material(material=material_id)
+    response.status_code = 200
+    return format_response(data=str(material_id))
```

Comment:
you always return Success no mater if the resource existed, or user had (/not) access to it

Reply:
Replaced with:

```python
    try:
        material_db.delete_material(material=material_id)
    except KeyError as e:
        message = str(e)
        response.status_code = 400
        return format_response(message=message)
```